domains/client/block-preprocessor/src/lib.rs

```diff
@@ -331,23 +359,23 @@ where
         >,
         consensus_block_hash: CBlock::Hash,
         at: Block::Hash,
-    ) -> sp_blockchain::Result<bool> {
+    ) -> Result<bool, sp_api::ApiError> {
```

Comment:
If we return `Option<Vec<Block::Extrinsic>>` instead of `bool` then we won't need to decode the extrinsic a second time, it may hurt readability though, non-blocker.

Reply:
Yeah, thought about avoiding the decoding again, but unsure if it's worth the readability comprise.